crypto/cloudkms/cloudkms.go

```diff
@@ -57,6 +60,25 @@ func (k Key) ResourceID() string {
 	)
 }
 
+func KeyFromResourceID(resourceID string) (Key, error) {
+	key := Key{}
+
+	scanned, err := fmt.Sscanf(
+		strings.ReplaceAll(resourceID, "/", " "),       // input
+		strings.ReplaceAll(resourceIDFormat, "/", " "), // format
+		&key.ProjectID, &key.LocationID, &key.KeyRingID, &key.KeyID, &key.KeyVersion, // arugments to fill
+	)
+
+	if err != nil {
+		fmt.Println(resourceID, resourceIDFormat, key.ProjectID, key.LocationID, key.KeyRingID, key.KeyID, key.KeyVersion)
+		return key, fmt.Errorf("Could not scanf %s: %w", resourceID, err)
+	} else if scanned != 5 {
```

Comment:
I'd just put this in its own block since it's not related to the error return case above 
```suggestion
	}
	
	if scanned != 5 {
```